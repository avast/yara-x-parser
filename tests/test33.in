//Rule comment
rule test
{
	//Rule block comment
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		(1+1) of ($a, $b*) and $b in (100..filesize) or 9 < filesize
}
