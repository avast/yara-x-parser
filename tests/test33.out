SOURCE_FILE@0..162
  RULE@0..161
    COMMENT@0..14 "//Rule comment"
    WHITESPACE@14..15 "\n"
    RULE_KW@15..19 "rule"
    WHITESPACE@19..20 " "
    IDENTIFIER@20..24 "test"
    WHITESPACE@24..25 "\n"
    BLOCK_EXPR@25..161
      L_BRACE@25..26 "{"
      WHITESPACE@26..28 "\n\t"
      STRINGS@28..84
        COMMENT@28..48 "//Rule block comment"
        WHITESPACE@48..50 "\n\t"
        STRINGS_KW@50..57 "strings"
        COLON@57..58 ":"
        WHITESPACE@58..61 "\n\t\t"
        VARIABLE_STMT@61..71
          VARIABLE@61..63 "$a"
          WHITESPACE@63..64 " "
          ASSIGN@64..65 "="
          WHITESPACE@65..66 " "
          PATTERN@66..71
            STRING_LIT@66..71 "\"foo\""
        WHITESPACE@71..74 "\n\t\t"
        VARIABLE_STMT@74..84
          VARIABLE@74..76 "$b"
          WHITESPACE@76..77 " "
          ASSIGN@77..78 "="
          WHITESPACE@78..79 " "
          PATTERN@79..84
            STRING_LIT@79..84 "\"bar\""
      WHITESPACE@84..86 "\n\t"
      CONDITION@86..159
        CONDITION_KW@86..95 "condition"
        COLON@95..96 ":"
        WHITESPACE@96..99 "\n\t\t"
        BOOLEAN_EXPR@99..159
          BOOLEAN_EXPR@99..143
            BOOLEAN_TERM@99..117
              OF_EXPR@99..117
                QUANTIFIER@99..104
                  PRIMARY_EXPR@99..104
                    L_PAREN@99..100 "("
                    EXPRESSION@100..103
                      PRIMARY_EXPR@100..101
                        INT_LIT@100..101 "1"
                      PLUS@101..102 "+"
                      PRIMARY_EXPR@102..103
                        INT_LIT@102..103 "1"
                    R_PAREN@103..104 ")"
                WHITESPACE@104..105 " "
                OF_KW@105..107 "of"
                WHITESPACE@107..108 " "
                PATTERN_IDENT_TUPLE@108..117
                  L_PAREN@108..109 "("
                  VARIABLE_WILDCARD@109..111
                    VARIABLE@109..111 "$a"
                  COMMA@111..112 ","
                  WHITESPACE@112..113 " "
                  VARIABLE_WILDCARD@113..116
                    VARIABLE@113..115 "$b"
                    STAR@115..116 "*"
                  R_PAREN@116..117 ")"
            WHITESPACE@117..118 " "
            AND_KW@118..121 "and"
            WHITESPACE@121..122 " "
            BOOLEAN_EXPR@122..143
              BOOLEAN_TERM@122..143
                VARIABLE@122..124 "$b"
                WHITESPACE@124..125 " "
                VARIABLE_ANCHOR@125..143
                  IN_KW@125..127 "in"
                  WHITESPACE@127..128 " "
                  RANGE@128..143
                    L_PAREN@128..129 "("
                    PRIMARY_EXPR@129..132
                      INT_LIT@129..132 "100"
                    DOTDOT@132..134 ".."
                    PRIMARY_EXPR@134..142
                      FILESIZE_KW@134..142 "filesize"
                    R_PAREN@142..143 ")"
          WHITESPACE@143..144 " "
          OR_KW@144..146 "or"
          WHITESPACE@146..147 " "
          BOOLEAN_EXPR@147..159
            BOOLEAN_TERM@147..159
              EXPRESSION@147..159
                PRIMARY_EXPR@147..148
                  INT_LIT@147..148 "9"
                WHITESPACE@148..149 " "
                LT@149..150 "<"
                WHITESPACE@150..151 " "
                PRIMARY_EXPR@151..159
                  FILESIZE_KW@151..159 "filesize"
      WHITESPACE@159..160 "\n"
      R_BRACE@160..161 "}"
  WHITESPACE@161..162 "\n"
